test/gtest/ucp/test_ucp_worker.cc

```diff
@@ -928,6 +930,52 @@ UCS_TEST_P(test_pci_bw, get_pci_bw)
 
 UCP_INSTANTIATE_TEST_CASE_TLS(test_pci_bw, all, "all")
 
+class rlimit_setter {
+public:
+    rlimit_setter() : m_resource(0), m_limit(0), m_orig({})
+    {
```

Comment:
I think we need to save original rlimit in the ctor (and only here).
Because subsequent set calls may override the stored value, and we never get back to original

Reply:
there might also be multiple instantiation of rlimit_setter, released in different order